cwl/src/main/scala/cwl/CwlType.scala

```diff
@@ -235,12 +235,8 @@ case class Directory private
 
   lazy val asWomValue: ErrorOr[WomFile] = {
     errorOrListingOption flatMap { listingOption =>
-      val valueOption = path.orElse(location).orElse(basename)
-      (valueOption, listingOption) match {
-        case (None, None) =>
-          "Cannot convert CWL File to WomValue without either a location, a path, a basename, or a listing".invalidNel
-        case (_, _) => WomMaybeListedDirectory(valueOption, listingOption).valid
-      }
+      val valueOption = path.orElse(location).orElse(basename).orElse(Option(better.files.File.newTemporaryDirectory().pathAsString))
```

Comment:
This leaks files (technically directories), and is works as a temporary patch for SFS only AFAICT.

Again, similar to Dan's PR I'm fine with it as long as it's tracked.

Reply:
“Tracking” generally leads to “forgotten”

Reply:
Conformance 115/116 won't be happy on PAPI until this is fixed there which will warrant a more general and hopefully less horrible solution.